internal/ethapi/api.go

```diff
@@ -1062,17 +1063,17 @@ func doCall(ctx context.Context, b Backend, args TransactionArgs, state *state.S
 	gp := new(core.GasPool).AddGas(math.MaxUint64)
 	result, err := core.ApplyMessage(evm, msg, gp)
 	if err := vmError(); err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	// If the timer caused an abort, return an appropriate error message
 	if evm.Cancelled() {
-		return nil, fmt.Errorf("execution aborted (timeout = %v)", timeout)
+		return nil, 0, fmt.Errorf("execution aborted (timeout = %v)", timeout)
 	}
 	if err != nil {
-		return result, fmt.Errorf("err: %w (supplied gas %d)", err, msg.GasLimit)
+		return result, 0, fmt.Errorf("err: %w (supplied gas %d)", err, msg.GasLimit)
 	}
-	return result, nil
+	return result, math.MaxUint64 - gp.Gas(), nil
```

Comment:
I don't think it is great using `math.MaxUint64` here with the requirement it is the starting point for the gas pool. But more importantly, I don't see why the function signature needs to change here. `result` has `gasUsed` as a field. Is this not an acceptable way of determining the amount of gas used on the call?

Reply:
yeah agree, I somehow missed the UsedGas value right there in the result :man_facepalming: 
fixed